code/modules/atmospherics/environmental/LINDA_turf_tile.dm

```diff
@@ -199,30 +209,43 @@
 			our_excited_group = excited_group
 			should_share_air = TRUE
 
-		//air sharing
-		if(should_share_air)
-			var/difference = our_air.share(enemy_air, adjacent_turfs_length)
-			if(difference)
-				if(difference > 0)
-					consider_pressure_difference(enemy_tile, difference)
-				else
-					enemy_tile.consider_pressure_difference(src, -difference)
-			LAST_SHARE_CHECK
+	if(should_share_air)
+		var/datum/gas_mixture/final_air = new()
+		final_air.merge(our_air.copy())
+		for(var/t in adjacent_turfs)
+			var/turf/open/enemy_tile = t
+			final_air.merge(enemy_tile.air.copy())
+		if (planet_atmos) //share our air with the "atmosphere" "above" the turf
+			var/datum/gas_mixture/G = new
+			G.copy_from_turf(src)
+			ARCHIVE_TEMPERATURE(G)
+			if(!(our_air.compare(G)))
```

Comment:
Are the parethesis after the ! operator really needed?

Reply:
they don't hurt for sure.